packages/perseus/src/widget-type-utils.test.ts

```diff
@@ -0,0 +1,271 @@
+import {
+    generateTestCategorizerWidget,
+    generateTestPerseusItem,
+    generateTestRadioWidget,
+} from "./util/test-utils";
+import {
+    getWidgetTypeByWidgetId,
+    contentHasWidgetType,
+    getWidgetIdsFromContent,
+    getWidgetIdsFromContentByType,
+    getWidgetsMapFromItemData,
+    getWidgetFromWidgetMap,
+    getWidgetsFromWidgetMap,
+} from "./widget-type-utils";
+
+describe("widget-type-utils", () => {
+    describe("getWidgetTypeByWidgetKey", () => {
+        it("returns widget type when found", () => {
+            // Assemble
+            const widgetId = "dont-look-for-type-in-key";
+            const widgetMap = {
+                [widgetId]: generateTestRadioWidget(),
+            };
+
+            // Act
+            const widgetType = getWidgetTypeByWidgetId(widgetId, widgetMap);
+
+            // Assert
+            expect(widgetType).toBe("radio");
+        });
+
+        it("returns null when not found", () => {
+            // Assemble
+            const widgetId = "dont-look-for-type-in-key";
+            const widgetMap = {};
+
+            // Act
+            const widgetType = getWidgetTypeByWidgetId(widgetId, widgetMap);
+
+            // Assert
+            expect(widgetType).toBeNull();
+        });
+    });
+
+    describe("contentHasWidgetType", () => {
+        it("returns true when found", () => {
+            // Assemble
+            const type = "radio";
+            const widgetId = "dont-look-for-type-in-key";
+            const content = `[[☃ ${widgetId}]]`;
+            const widgetMap = {
+                [widgetId]: generateTestRadioWidget(),
+            };
+
+            // Act
+            const result = contentHasWidgetType(type, content, widgetMap);
+
+            // Assert
+            expect(result).toBeTrue();
+        });
+
+        it("returns false when found", () => {
```

Comment:
Comment, nb: The names for the tests are a little confusing. I like how short they are, but it might be better to describe what the test is doing a bit more.

Reply:
I tend to think of a test name as being a combination of the describe + test, in this case: "contentHasWidgetType returns false when found" which covers:

- The name of the function: "contentHasWidgetType"
- What we're searching: "content"
- What we're looking for: "widget type"
- The assertion aka the expected return value: "false" (also "Has" implies a boolean return type)

IMO: I think that + focused functions + focused tests should be enough here unless you think there's something specific that we're missing.

Reply:
I think it just sounds confusing to say that the function returns false when it finds the widget type. It returns false when the widget type is _not_ found, right?

Reply:
I see what you mean. I thought you were talking more generally about the names. This one in particular is wrong, so I fixed it. Thanks for pointing that out.